test_particle_in_field.py

```diff
@@ -38,3 +38,11 @@ def test_chargedparticle(sim_example):
     assert isinstance(charged_example, ChargedParticle)
     assert charged_example.E is None
     assert charged_example.x == particle_data["position"]
+
+
+def test_id_of_shared_attribute_should_not_change_after_update(sim_example):
+    """Tests if a shared attribute references the same object after update"""
+    before = EMWave(sim_example, wave_data)
```

Comment:
Can you change this variable name to something like `wave`? I know it's not important, but calling it `before` makes the test "read" strangely... `id(before.E)` sounds like it should be the id of E from before the update was called.